From ceaaa4f44f9b3ec82c4e0a24c2322aae58fa3aa0 Mon Sep 17 00:00:00 2001
From: Joachim Wiberg <troglobit@gmail.com>
Date: Wed, 6 Nov 2024 15:39:33 +0100
Subject: [PATCH 34/36] net: dsa: mb88e6xxx: use EOPNOTSUPP for unsupported
 flags
Organization: Wires

Make sure to return correct error code for unsupported flags, and
propagate any error.

Follow-up to a5cf5c0

Signed-off-by: Joachim Wiberg <troglobit@gmail.com>
Signed-off-by: Tobias Waldekranz <tobias@waldekranz.com>
---
 drivers/net/dsa/mv88e6xxx/leds.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/dsa/mv88e6xxx/leds.c b/drivers/net/dsa/mv88e6xxx/leds.c
index b2a55d96b506..fe1ed3ed2a8a 100644
--- a/drivers/net/dsa/mv88e6xxx/leds.c
+++ b/drivers/net/dsa/mv88e6xxx/leds.c
@@ -99,7 +99,7 @@ static int mv88e6393x_led_flags_to_mode(struct mv88e6xxx_led *led, unsigned long
 			return i;
 	}
 
-	return -EINVAL;
+	return -EOPNOTSUPP;
 }
 
 static int mv88e6393x_led_mode_to_flags(struct mv88e6xxx_led *led, u8 mode,
@@ -216,6 +216,9 @@ static int mv88e6393x_led_hw_control_set(struct mv88e6xxx_led *led,
 {
 	int mode = mv88e6393x_led_flags_to_mode(led, flags);
 
+	if (mode < 0)
+		return mode;
+
 	return mv88e6393x_led_set(led, mode);
 }
 
-- 
2.43.0

